File: hw/calle/u2/hw19Inheritance/FarmsystemPython/farmsystem_model/pig.py
from datetime import date
from .farm_animal import FarmAnimal
from .cage import Cage
from typing import Self

class Pig(FarmAnimal):
    def __init__(self, id: int, breed: str, born_on: date, gender: str, is_able_to_reproduce: bool, weight: float, cage: Cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)

    def __str__(self) -> str:
        return f"Pig{{{super().__str__()} }}"

File: hw/santorum/u2/hw19Inheritance/Java/Farmsystem/src/main/java/ec/edu/espe/farmsystem/model/Pig.java

package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author Windows
 */

public class Pig extends FarmAnimal {
    private float foodPerDay;
    private boolean isReadyForSale;

    public Pig(float foodPerDay, boolean isReadyForSale, int id, String breed, Date bornOn,
               String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.foodPerDay = foodPerDay;
        this.isReadyForSale = isReadyForSale;
    }

    public void feed(float amount) {
        foodPerDay += amount;
    }

    @Override
    public String toString() {
        return "Pig{" +
                "foodPerDay=" + foodPerDay +
                ", isReadyForSale=" + isReadyForSale +
                ", " + super.toString() +
                '}';
    }

    public float getFoodPerDay() {
        return foodPerDay;
    }

    public void setFoodPerDay(float foodPerDay) {
        this.foodPerDay = foodPerDay;
    }

    public boolean isReadyForSale() {
        return isReadyForSale;
    }

    public void setReadyForSale(boolean readyForSale) {
        isReadyForSale = readyForSale;
    }
}
